src/supertux/autotile.cpp

```diff
@@ -77,11 +77,22 @@ Autotile::pick_tile(int x, int y) const
 
   for (const auto& pair : m_alt_tiles)
   {
-    rnd_val -= pair.second;
-    if (rnd_val <= 0)
+    const AltConditions& cond = pair.second;
+    if (cond.weight <= 0.f && !cond.period_x.first && !cond.period_y.first)
+      continue;
+
+    if (cond.period_x.first && x % cond.period_x.first != cond.period_x.second)
```

Comment:
What do you want by checking for only `cond.period_x.first`? Checking if it exists? If it isn't zero?

Reply:
Yes, if it isn't 0.

Reply:
Can you write that explicitly?
```suggestion
    if (cond.period_x.first != 0 && x % cond.period_x.first != cond.period_x.second)
```

(in the other conditions too)

Reply:
Why? It cannot be mistaken for anything else, can it?

Reply:
it can, and I did, because i'm used to seeing that only in booleans and pointers. i just suggested that to save some time reading that. not a big deal